React/CoreModules/RCTDevLoadingView.mm

```diff
@@ -266,16 +266,21 @@ - (void)showProgressMessage:(NSString *)message
 
 - (void)showOfflineMessage
 {
-  RCTUIColor *color = [RCTUIColor whiteColor]; // TODO(macOS GH#774)
-  RCTUIColor *backgroundColor = [RCTUIColor blackColor]; // TODO(macOS GH#774)
-
-  if ([self isDarkModeEnabled]) {
-    color = [RCTUIColor blackColor]; // TODO(macOS GH#774)
-    backgroundColor = [RCTUIColor whiteColor]; // TODO(macOS GH#774)
-  }
+  // [TODO(macOS GH#774) - isDarkModeEnabled should only be run on the main thread
+  __weak __typeof(self) weakSelf = self;
+  RCTExecuteOnMainQueue(^{
+    RCTUIColor *color = [RCTUIColor whiteColor]; // TODO(macOS GH#774)
+    RCTUIColor *backgroundColor = [RCTUIColor blackColor]; // TODO(macOS GH#774)
+
+    if ([weakSelf isDarkModeEnabled]) {
+      color = [RCTUIColor blackColor]; // TODO(macOS GH#774)
+      backgroundColor = [RCTUIColor whiteColor]; // TODO(macOS GH#774)
+    }
```

Comment:
I would still overall just vote we replace these colors with RCTDynamicColor(light:_, dark:_) and try to push that change upstream. I don't think we should be checking `isDarkModeEnabled` at all, or at least if so, ensure the caller is on the main thread to begin with

Reply:
I feel that for the sake of code cleanliness, we should at least wait until RCTDynamicColor is available on iOS as well before switching patterns (after all, we might as well do this for all instances in this file). I'm not sure why it's only available for macOS now, but I think that the work for doing this is a bit out of scope here.

Reply:
It's available on iOS, it's just not a custom class. I would take a look at `RCTConvert.m` and look for the `UIColor` method to see how iOS dynamic colors are made.

Reply:
https://github.com/microsoft/react-native-macos/blob/3bcd858a89062d3a77f88e81e1bf6c7624c43794/React/Base/RCTConvert.m#L1051

Reply:
Talked offline, this is probably OK to unblock with minimal diffs. I would prefer a better solution pushed upstream but this works.

Reply:
> Talked offline, this is probably OK to unblock with minimal diffs. I would prefer a better solution pushed upstream but this works.

Yeah, we can probably push a better fix upstream at a later point, but for now I'll resolve this thread.